<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="tab1">
   <Container>
    <DockSplitter orientation="-" count="2" sizes="0.5;0.5">
     <DockSplitter orientation="|" count="2" sizes="0.5;0.5">
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range bottom="-266.538895" left="10.680400" right="30.680400" top="-249.999397"/>
        <limitY/>
        <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/er" color="#1f77b4"/>
       </plot>
      </DockArea>
      <DockSplitter orientation="-" count="3" sizes="0.334066;0.331868;0.334066">
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-2.775037" left="10.680400" right="30.680400" top="268.097421"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/ek_indct.0" color="#1ac938"/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/ek_indct.1" color="#ff7f0e"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-2.775037" left="10.680400" right="30.680400" top="268.097421"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/ek_joule.0" color="#f14cc1"/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/ek_joule.1" color="#9467bd"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-2.775037" left="10.680400" right="30.680400" top="268.097421"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/ek_mech.0" color="#17becf"/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/ek_mech.1" color="#bcbd22"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
     <DockSplitter orientation="|" count="2" sizes="0.5;0.5">
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range bottom="-0.801297" left="10.680400" right="30.680400" top="-0.774067"/>
        <limitY/>
        <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pr" color="#d62728"/>
       </plot>
      </DockArea>
      <DockSplitter orientation="-" count="3" sizes="0.334066;0.331868;0.334066">
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-0.000000" left="10.680400" right="30.680400" top="0.000000"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pk_indct.0" color="#1f77b4"/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pk_indct.1" color="#d62728"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="0.018473" left="10.680400" right="30.680400" top="0.782055"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pk_joule.0" color="#1ac938"/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pk_joule.1" color="#ff7f0e"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-0.000030" left="10.680400" right="30.680400" top="0.000149"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pk_mech.0" color="#f14cc1"/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pk_mech.1" color="#9467bd"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter orientation="-" count="2" sizes="0.5;0.5">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range bottom="-268.143544" left="10.680400" right="30.680400" top="2.779885"/>
       <limitY/>
       <curve name="/bus_power_rt/reg_pow_node/er_k.0" color="#1f77b4"/>
       <curve name="/bus_power_rt/reg_pow_node/er_k.1" color="#d62728"/>
       <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/er_k.0" color="#17becf"/>
       <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/er_k.1" color="#bcbd22"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range bottom="-0.782204" left="10.680400" right="30.680400" top="-0.018445"/>
       <limitY/>
       <curve name="/bus_power_rt/reg_pow_node/pr_k.0" color="#1ac938"/>
       <curve name="/bus_power_rt/reg_pow_node/pr_k.1" color="#ff7f0e"/>
       <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pr_k.0" color="#1f77b4"/>
       <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pr_k.1" color="#d62728"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab3">
   <Container>
    <DockSplitter orientation="-" count="3" sizes="0.332968;0.334064;0.332968">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range bottom="-0.000000" left="10.680400" right="30.680400" top="0.000000"/>
       <limitY/>
       <curve name="/bus_power_rt_iq_model/iq_est_node_pow_iq_model/q_p_ddot_est_filt.0" color="#1ac938"/>
       <curve name="/bus_power_rt_iq_model/iq_est_node_pow_iq_model/q_p_ddot_est_filt.1" color="#ff7f0e"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range bottom="-0.000022" left="10.680400" right="30.680400" top="0.000028"/>
       <limitY/>
       <curve name="/bus_power_rt_iq_model/iq_est_node_pow_iq_model/q_p_dot_meas.0" color="#f14cc1"/>
       <curve name="/bus_power_rt_iq_model/iq_est_node_pow_iq_model/q_p_dot_meas.1" color="#9467bd"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range bottom="0.474625" left="10.680400" right="30.680400" top="2.425166"/>
       <limitY/>
       <curve name="/bus_power_rt_iq_model/iq_est_node_pow_iq_model/iq_est.0" color="#17becf"/>
       <curve name="/bus_power_rt_iq_model/iq_est_node_pow_iq_model/iq_est.1" color="#bcbd22"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter orientation="-" count="1" sizes="1">
     <DockSplitter orientation="|" count="2" sizes="0.30732;0.69268">
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range bottom="-0.100000" left="10.680400" right="30.680400" top="0.100000"/>
        <limitY/>
        <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/recov_energy_tot" color="#1ac938"/>
       </plot>
      </DockArea>
      <DockSplitter orientation="-" count="4" sizes="0.25;0.25;0.25;0.25">
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-0.100000" left="10.680400" right="30.680400" top="0.100000"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/recov_energy.0" color="#1f77b4"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-0.764791" left="10.680400" right="30.680400" top="-0.732380"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pr_k.0" color="#1f77b4"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-0.100000" left="10.680400" right="30.680400" top="0.100000"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/recov_energy.1" color="#d62728"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range bottom="-0.041687" left="10.680400" right="30.680400" top="-0.036506"/>
         <limitY/>
         <curve name="/bus_power_rt_iq_model/reg_pow_node_iq_model/pr_k.1" color="#d62728"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="3"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis="" delimiter="0"/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

